lib/ancestry/has_ancestry.rb

```diff
@@ -86,7 +86,7 @@ def has_ancestry options = {}
       after_destroy :touch_ancestors_callback
 
       if ActiveRecord::VERSION::STRING >= '5.1.0'
-        after_save :touch_ancestors_callback, if: :saved_changes?
+        after_save :touch_ancestors_callback, if: :has_changes_to_save?
```

Comment:
`saved_changes?` and `has_changes_to_save?` have different semantics. `saved_changes?` is true if there were any changes when an instance was saved last time. `has_changes_to_save?` (which is an alias to `changed?`) is true if there are _unsaved_ changes. 

With this change touch_ancestors_callback is never called, two more tests fail. Tested with AR 5.2.0.

Reply:
@khustochka thanks for the insight. It sounds like `save_changes?` is the one we till want to use